src/tree/decorators.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  Component,
+  DebugElement,
+} from '@angular/core';
```

Comment:
@clbond is this used on the backend? If so I'd like us to avoid importing anything from angular, or any other libraries.